internal/consumer/consumer.go

```diff
@@ -289,6 +237,80 @@ func (i *InventoryConsumer) Consume() error {
 	return err
 }
 
+func (i *InventoryConsumer) ProcessMessage(headers map[string]string, relationsEnabled bool, msg *kafka.Message) (string, error) {
+	operation := headers["operation"]
+	txid := headers["txid"]
+
+	switch operation {
+	case string(model.OperationTypeCreated):
+		i.Logger.Infof("operation=%s tuple=%s txid=%s", operation, msg.Value, txid)
+		if relationsEnabled {
+			tuple, err := ParseCreateOrUpdateMessage(msg.Value)
+			if err != nil {
+				i.Logger.Errorf("failed to parse message for tuple: %v", err)
```

Comment:
Should these blocks be returning errors? Or will that conflict with the retry/restart logic?

Reply:
returns added